### PR TITLE
Mm 42775 create client 4 method for fetching trial self hosted

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR creates a Client4 method for requesting the trial license without having to call explicitly the doFetchWithResponse method, which should be private. This PR also makes private the Client4.doFetch and Client4.doFetchWithResponse

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42775

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: packages/client/src/client4.ts [hunk 1]
```diff
@@ -3863,13 +3871,13 @@ export default class Client4 {
 
     // Client Helpers
 
-    doFetch = async <ClientDataResponse>(url: string, options: Options): Promise<ClientDataResponse> => {
+    private doFetch = async <ClientDataResponse>(url: string, options: Options): Promise<ClientDataResponse> => {
```
Comment (JulienTant) [inline packages/client/src/client4.ts:3874]: Yes!!!

**Review by JulienTant — APPROVED** [review]
Looks good @pvev

**Review by hmhealey — APPROVED** [review]
Thanks! I really need to remember we have visibility modifiers now since JS has always made encapsulation miserable

**Comment by stevemudie** [comment]
@pvev https://automation-dashboard.vercel.app/cycles/63d28f14-15eb-4b4a-8d62-7ec29661c1f3

**Comment by pvev** [comment]
> @pvev https://automation-dashboard.vercel.app/cycles/63d28f14-15eb-4b4a-8d62-7ec29661c1f3

lol, e2e hate me! no idea why this simple change could potentially break the 100% of the e2e tests :) . Let me check what is going on with it. Thanks @stevemudie

**Comment by Phrynobatrachus** [comment]
I just killed the last e2e run, all but 1 of the runners crashed so it would take another 2-3 hours for the tests to complete.

restarting :crossed_fingers: 

/e2e-test

**Comment by pvev** [comment]
> I just killed the last e2e run, all but 1 of the runners crashed so it would take another 2-3 hours for the tests to complete.
> 
> restarting 🤞
> 
> /e2e-test

thanks @Phrynobatrachus ... is me again 🤣

**Review by Phrynobatrachus — APPROVED** [review]
E2E run LGTM.

https://automation-dashboard.vercel.app/cycles/bf119b23-5326-41c8-853d-c1459015c875

**Pull request merged**